cusub_perception/detection_tree/include/detection_tree/detection_tree_node.hpp

```diff
@@ -0,0 +1,93 @@
+#ifndef DETECTION_TREE_NODE_SRC_DETECTION_TREE_NODE_H_
+#define DETECTION_TREE_NODE_SRC_DETECTION_TREE_NODE_H_
+
+// Logging Formats
+#include <string>
+#define DETECTION_TREE_NAME             std::string("[\033[92mDetection Tree\033[0m] ")
+#define DETECTION_TREE_WARN_START       std::string("\033[93m[WARN] ")
+#define DETECTION_TREE_COLOR_START      std::string("\033[95m")
+#define DETECTION_TREE_END              std::string("\033[0m")
+
+# define DOBJECT_PROBABILITY_ZERO       0.0
+# define DOBJECT_NOT_FOUND              -1
+
+#include <pluginlib/class_list_macros.h>
+#include "detection_tree/detection_tree.hpp"
+#include <nodelet/nodelet.h>
+#include <ros/ros.h>
+#include <ros/timer.h>
+#include <vector>
+#include <map>
+#include <string>
+#include <cmath>
+#include <darknet_ros_msgs/BoundingBoxes.h>
+#include <darknet_ros_msgs/BoundingBox.h>
+#include <std_msgs/Header.h>
+#include <geometry_msgs/PoseStamped.h>
+#include <geometry_msgs/PoseArray.h>
+#include <geometry_msgs/Quaternion.h>
+#include <tf/transform_listener.h>
+#include <tf/transform_datatypes.h>
+#include <tf2/LinearMath/Quaternion.h>
+#include <tf2_geometry_msgs/tf2_geometry_msgs.h>
+#include <sensor_msgs/CameraInfo.h>
+#include <cv_bridge/cv_bridge.h>
+#include <sensor_msgs/image_encodings.h>
+#include <opencv2/calib3d.hpp>
+#include <detection_tree/Dvector.h>
+
+// Camel case classes + functions, underscore variables + namespaces
+
+namespace det_tree_ns
+{
+    class DetectionTree : public nodelet::Nodelet
+    {
+        public:
+            virtual void onInit();
+        private:
+            std::string sub_name;
+            void darknetCallback(const darknet_ros_msgs::BoundingBoxesConstPtr bbs);
+            void cameraInfoCallback(const sensor_msgs::CameraInfo ci);
+            int transformBearingToOdom(geometry_msgs::PoseStamped& odom_cam_pose, cv::Mat& bearing_vec, std_msgs::Header& image_header);
+            int createDobject(detection_tree::Dvector* dv);
+            bool poseSolveDobject(Dobject* dobj, geometry_msgs::Pose& pose);
+            void dobjPubCallback(const ros::TimerEvent&);
+            void det_print(std::string str);
+            void det_print_warn(std::string str);
+
+            // Dvector Association Functions
+            void associateDvectors(std::vector<detection_tree::Dvector*>& dv_list, std::map<detection_tree::Dvector*, int>& dv_dobj_map);            
+            void assignDobjScores(std::vector<detection_tree::Dvector*>& dv_list, std::map<detection_tree::Dvector*, std::map<int, double>*>& dvs_scored);
+            void setDobjProbabilityToZero(std::map<detection_tree::Dvector*, std::map<int, double>*>& dvs_scored, int dobj_num);
+            detection_tree::Dvector* findBestMatch(std::map<detection_tree::Dvector*, std::map<int, double>*>& dvs_scored, int& matching_dobj);
```

Comment:
At first glance, I was thinking why not just use class name, but to me, this way seems to fix using same name for different objects (i.e. start gate poles) as well as misdetection issues. Are those the motivation, or am I missing something?

Reply:
Exacltly, class name would be the preferable way but we have a few classes where more than one instantiation exists (3 start gate poles, multiple coffins), so the concept of grouping detections by the actual detection source is needed. All of the these functions are figuring out which detection source triggered the detection